src/drivers/imx/sai.c

```diff
@@ -251,6 +255,9 @@ static inline int sai_set_config(struct dai *dai, struct ipc_config_dai *common_
 		clk_div = (config->sai.mclk_rate / config->sai.bclk_rate / 2) - 1;
 	}
 
+	/* TDM mode is enabled only when fmt is dsp_a or dsp_b */
+	tdm_mode = TDM_DISABLE;
+
 	switch (config->format & SOF_DAI_FMT_FORMAT_MASK) {
 	case SOF_DAI_FMT_I2S:
```

Comment:
Here no need to enable tdm_mode?

Reply:
The reasoning here is like that: when user wants to use TDM mode (e.g up to 32 channels) they will set config->format to dsp_a and dsp_b. 

This way DSP will now that we will have more than 2 channels.

I think this is a little bit confusing but it is a convention between Linux kernel and dsp firmware. 

We can refine this code in the future.

Reply:
Got it, then ok for me.